iroha-cli/interactive/interactive_query_cli.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_CLI_INTERACTIVE_QUERY_CLI_HPP
+#define IROHA_CLI_INTERACTIVE_QUERY_CLI_HPP
+
+#include <unordered_map>
+#include "interactive/interactive_common_cli.hpp"
+#include "model/generators/query_generator.hpp"
+#include "model/query.hpp"
+
+namespace iroha_cli {
+  namespace interactive {
+    class InteractiveQueryCli {
+     public:
+      /**
+       * @param account_id creator's account identification
+       * @param query_counter counter associated with creator's account
+       */
+      InteractiveQueryCli(std::string account_id, uint64_t query_counter);
+      /**
+       * Run interactive query command line
+       */
+      void run();
+
+     private:
+
+      using QueryName = std::string;
+      using QueryParams = std::vector<std::string>;
+
+      // Menu for Queries
+      MenuPoints menu_points_;
+      /**
+       * Create query menu and assign command handlers for current class
+       * object
+       */
+      void create_queries_menu();
+
+      // Menu for Query result
+      MenuPoints result_points_;
+
+      /**
+       * Create result menu and assign result handlers for current class object
+       */
+      void create_result_menu();
+
+      // ------ Query handlers -----------
+      const std::string GET_ACC = "get_acc";
+      const std::string GET_ACC_AST = "get_acc_ast";
+      const std::string GET_ACC_TX = "get_acc_tx";
+      const std::string GET_ACC_SIGN = "get_acc_sign";
+
+      // ------  Query parsers ---------
+      using QueryHandler = std::shared_ptr<iroha::model::Query> (
+          InteractiveQueryCli::*)(QueryParams);
```

Comment:
Wow
what does `InteractiveQueryCli::*` in param means?

Reply:
Pointer to a member. I found a good example at so: https://stackoverflow.com/questions/9939305/what-is-in-c